File: web/src/api/orgs.ts
import request from './request'

export interface OrgResponse {
  org_id: string; name: string; description: string | null
  created_by: string; role: string | null; created_at: string | null; updated_at: string | null
}
export interface MemberResponse {
  membership_id: string; org_id: string; user_id: string
  username: string; display_name: string | null; role: string; joined_at: string | null
}
export interface CreateOrgRequest { name: string; description?: string }
export interface UpdateOrgRequest { name?: string; description?: string }
export interface InviteMemberRequest { username: string }
export interface ChangeRoleRequest { role: 'admin' | 'member' }

export const createOrg = (data: CreateOrgRequest) => request.post<any, OrgResponse>('/orgs', data)
export const listMyOrgs = () => request.get<any, OrgResponse[]>('/orgs')
export const getOrg = (orgId: string) => request.get<any, OrgResponse>(`/orgs/${orgId}`)
export const updateOrg = (orgId: string, data: UpdateOrgRequest) =>
  request.patch<any, OrgResponse>(`/orgs/${orgId}`, data)
export const listMembers = (orgId: string) =>
  request.get<any, MemberResponse[]>(`/orgs/${orgId}/members`)
export const inviteMember = (orgId: string, data: InviteMemberRequest) =>
  request.post<any, { invitation_id: string }>(`/orgs/${orgId}/invitations`, data)
export const changeMemberRole = (orgId: string, userId: string, data: ChangeRoleRequest) =>
  request.patch<any, MemberResponse>(`/orgs/${orgId}/members/${userId}`, data)
export const removeMember = (orgId: string, userId: string) =>
  request.delete<any, { detail: string }>(`/orgs/${orgId}/members/${userId}`)
export const leaveOrg = (orgId: string) =>
  request.delete<any, { detail: string }>(`/orgs/${orgId}/members/me`)

File: src/api/schemas/orgs.py
"""组织与成员管理请求/响应 Schema"""

from datetime import datetime

from pydantic import BaseModel, Field


class CreateOrgRequest(BaseModel):
    name: str = Field(..., min_length=1, max_length=256)
    description: str | None = Field(None, max_length=1024)


class UpdateOrgRequest(BaseModel):
    name: str | None = Field(None, min_length=1, max_length=256)
    description: str | None = Field(None, max_length=1024)


class OrgResponse(BaseModel):
    org_id: str
    name: str
    description: str | None = None
    created_by: str
    role: str | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None


class MemberResponse(BaseModel):
    membership_id: str
    org_id: str
    user_id: str
    username: str
    display_name: str | None = None
    role: str
    joined_at: datetime | None = None


class SwitchOrgRequest(BaseModel):
    org_id: str = Field(..., min_length=1)
